### PR TITLE
Add whitelist to collection design

### PR DESCRIPTION
**Pull request by wyuinche**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
1. no whitelist in collection designs.
2. only collection-creator can mint to the collection.

* **What is the new behavior (if this is a feature change)?**
1. you can register collection with whitelist.
2. whitelisted accounts can mint to collection instead of collection-creator.

* **Other information**:
collection-policy, collection-register, mint have been updated.

### CODE REVIEW

File: nft/collection/mint_process.go [hunk 1]
```diff
@@ -59,8 +59,17 @@ func (ipp *MintItemProcessor) PreProcess(
...
-		return errors.Errorf("sender must be collection creator; creator: %q", design.Creator().String())
+	} else if policy, ok := design.Policy().(CollectionPolicy); !ok {
+		return errors.Errorf("policy of design is not collection-policy; %q", design.Symbol())
+	} else if whites := policy.Whites(); !design.Creator().Equal(ipp.sender) {
+		for i := range whites {
+			if whites[i].Equal(ipp.sender) {
+				break
+			}
+			if i == len(whites)-1 {
+				return errors.Errorf("sender is not whitelisted, nor is collection creator; %q", ipp.sender)
```
Comment (soonkuk) [inline nft/collection/mint_process.go:70]: @wyuinche 
I think it's good that the error message informs only one situation. If these two situations are included together, it makes it difficult to find the exact cause when debugging later.

**Review by soonkuk — CHANGES_REQUESTED** [review]

File: nft/collection/mint_process.go (hunk 1 above)
Comment (wyuinche) [reply nft/collection/mint_process.go]: @soonkuk It has been changed so that only whitelisted accounts can mint nft.

**Review by soonkuk — APPROVED** [review]
LGTM!

File: nft/collection/mint_process.go (hunk 1 above)
Comment (soonkuk) [reply nft/collection/mint_process.go]: @wyuinche Ok I got it.

**Pull request merged**